Sources/BraintreeDropIn/BTPaymentSelectionViewController.m

```diff
@@ -490,8 +301,60 @@ - (CGFloat)tableView:(__unused UITableView *)tableView heightForRowAtIndexPath:(
 
 #pragma mark UITableViewDataSource
 
-- (NSInteger)tableView:(__unused UITableView *)tableView numberOfRowsInSection:(__unused NSInteger)section {
-    return [self.paymentOptionsData count];
+- (NSInteger)tableView:(__unused UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
+    if (section == 0 && self.hasVaultedPaymentMethods) {
+        return 1;
+    } else {
+        return self.paymentOptionsData.count;
+    }
+}
+
+- (CGFloat)tableView:(UITableView *)tableView heightForHeaderInSection:(NSInteger)section {
+    if (self.hasVaultedPaymentMethods) {
+        return 35;
+    } else {
+        return CGFLOAT_MIN; // hide the header
```

Comment:
For some reason, returning 0 causes the table view to render empty header views at the default height. If there aren't any vaulted payment methods, we don't want a header view at all. [SO](https://stackoverflow.com/a/40974547) says to return CGFLOAT_MIN, which seems to work. 🤷‍♀️